rust/ql/test/library-tests/sensitivedata/test.rs

```diff
@@ -0,0 +1,150 @@
+
+fn get_string() -> String { "string".to_string() }
+
+fn sink<T>(_: T) { }
+
+// --- tests ---
+
+struct MyStruct {
+	harmless: String,
+	password: String,
+	password_file_path: String,
+	password_enabled: String,
+}
+
+impl MyStruct {
+	fn get_certificate(&self) -> String { return get_string() }
+	fn get_certificate_url(&self) -> String { return get_string() }
+	fn get_certificate_file(&self) -> String { return get_string() }
+}
+
+fn get_password() -> String { get_string() }
+
+fn test_passwords(
+	password: &str, passwd: &str, my_password: &str, password_str: &str, pass_phrase: &str,
+	auth_key: &str, authenticationkey: &str, authKey: &str,
+	harmless: &str, encrypted_password: &str, password_hash: &str,
+	ms: &MyStruct
+) {
+	// passwords
+	sink(password); // $ sensitive=password
+	sink(passwd); // $ sensitive=password
+	sink(my_password); // $ sensitive=password
+	sink(password_str); // $ sensitive=password
+	sink(pass_phrase); // $ MISSING: sensitive=password
+	sink(auth_key); // $ MISSING: sensitive=password
+	sink(authenticationkey); // $ sensitive=password
+	sink(authKey); // $ sensitive=password
+
+	sink(ms); // $ MISSING: sensitive=password
```

Comment:
Why should this one be `sensitive=password`?

Reply:
Because `ms` is a `MyStruct` which contains a sensitive field `password`.  So if you transmitted the whole of `ms` plaintext you would be transmitting a password (and some other stuff) unencrypted.